DRAFT_Connectivity.Rmd

```diff
@@ -1,266 +1,280 @@
+  var distance = infrastruktur.rename('infrastruktur').reduceRegion({
+    reducer: ee.Reducer.mean(),
+    geometry: feature.geometry().buffer(bufferDistance),
+    scale: 10,  // Oppløsningen til datasettet (meters per piksel)
+    maxPixels: 1e13
+  }).get('infrastruktur');
+  return feature.set('mean_distance', distance);
+};
+
+// Velg områder med myr (myrinnhold > 80%) og bruk dette som geometri for analysen
+var omraderMedMyr = myr.gt(0.8).selfMask();
+
+// Hent geometriene for hvert område med myr
+var omraderGeometries = omraderMedMyr.reduceToVectors({
+  geometry: Map.getBounds(true),
+  scale: 10,  // Oppløsningen til datasettet (meters per piksel)
+  maxPixels: 1e13
+});
+
+// Bruk funksjonen calculateMeanDistance for å beregne gjennomsnittlig avstand til infrastruktur for hvert område med myr
+
+// For scenariet "Konnektivitet med infrastruktur" 
+var omraderMedGjennomsnittligAvstand = omraderGeometries.map(calculateMeanDistance);
+
+// Legg til lag som viser myrpolygonene og lag med gjennomsnittlig avstand til infrastruktur på kartet
+Map.addLayer(omraderMedMyr, {min: 0, max: 1, palette: ['white', 'green']}, 'Myrpolygoner');
+
+// Vis scenariet "Konnektivitet med infrastruktur" på kartet
+var visParams = {
+  min: 0,
+  max: 1, // Juster dette basert på din infrastrukturindeksens min og maks
+  palette: ['blue', 'yellow', 'red'],
+};
+Map.addLayer(omraderMedGjennomsnittligAvstand, visParams, 'Gjennomsnittlig avstand til infrastruktur');
+
+// Eksporter resultatene som en Shapefile til Google Drive
+Export.table.toDrive({
+  collection: omraderMedGjennomsnittligAvstand,
+  description:'Konnektivitet_med_infrastruktur', // Navnet på eksportfilen for scenariet "Konnektivitet med infrastruktur"
+  fileFormat: 'SHP' // Filformat for eksport (Shapefile)
+});
+
+// Skift til scenariet "Konnektivitet uten infrastruktur" ved å fjerne infrastrukturdataene
+infrastruktur = ee.Image(0); // Nullstiller infrastruktur til et tomt bilde
+var omraderMedGjennomsnittligAvstandUtenInfrastruktur = omraderGeometries.map(calculateMeanDistance);
+
+// Vis scenariet "Konnektivitet uten infrastruktur" på kartet
+Map.addLayer(omraderMedGjennomsnittligAvstandUtenInfrastruktur, visParams, 'Gjennomsnittlig avstand uten infrastruktur');
+
+// Eksporter resultatene som en Shapefile til Google Drive for scenariet "Konnektivitet uten infrastruktur"
+Export.table.toDrive({
+  collection: omraderMedGjennomsnittligAvstandUtenInfrastruktur,
+  description:'Konnektivitet_uten_infrastruktur', // Navnet på eksportfilen for scenariet "Konnektivitet uten infrastruktur"
+  fileFormat: 'SHP' // Filformat for eksport (Shapefile)
+});
+
+
+```
+
+The scaled "Connectivity" values are created by dividing the result without infrastructure by the result that includes infrastructure. This ratio will give you a value between 0 and 1, where 0 represents complete disruption of connectivity due to infrastructure, and 1 represents undisturbed natural connectivity.
```

Comment:
Changes from _reference value_ to _scaled value_.

Reply:
I've made some updates to the code. There were indeed some mistakes. Please find the revised code below:

Thanks for seeing this!

```javascript
// Senter kartet til en bestemt geografisk posisjon (bredde- og lengdegrad) og velg en zoomnivå
Map.setCenter(11.0534047, 62.14137, 12);

// Laste inn infrastrukturindeks og myr datasett fra Google Earth Engine Data
var infrastruktur = ee.Image('users/vegar/NY_INFRA_IND');
var myrpred2 = ee.ImageCollection('users/vegarbakkestuen/Myr153');
var myr = myrpred2.mean();

// Legg til myrdata som et lag på kartet med tilpassede visningsparametere
Map.addLayer(myr, { min: 0, max: 1, palette: ['white', 'green'] }, 'myr');
Map.addLayer(infrastruktur, { min: 0, max: 13, palette: ['blue', 'yellow', 'red'] }, 'infrastruktur');

// Definer en bufferavstand (i meter) som bestemmer området som blir analysert rundt hvert myrpolygon
var bufferDistance = 500;

// Funksjon for å beregne gjennomsnittlig avstand til infrastruktur basert på grenseverdien
var calculateMeanDistance = function (feature) {
  var threshold = 3;  // Definer din grenseverdi for infrastruktur

  // Filtrer infrastrukturbildet basert på grenseverdien
  var filteredInfrastructure = infrastruktur.gt(threshold);

  // Hent myrpolygonets sentrum
  var myrCenter = feature.geometry().centroid(100);

  // Beregn gjennomsnittlig avstand til infrastruktur ved å redusere regionen rundt hvert myrpolygon
  var distanceWithInfrastructure = filteredInfrastructure.rename('filtered_infra').reduceRegion({
    reducer: ee.Reducer.mean(),
    geometry: feature.geometry().buffer(bufferDistance),
    scale: 10,  // Oppløsningen til datasettet (meters per piksel)
    maxPixels: 1e13
  }).get('filtered_infra');

  var distanceWithoutInfrastructure = filteredInfrastructure.rename('filtered_infra').reduceRegion({
    reducer: ee.Reducer.mean(),
    geometry: myrCenter.buffer(bufferDistance),
    scale: 10,  // Oppløsningen til datasettet (meters per piksel)
    maxPixels: 1e13
  }).get('filtered_infra');

  // Skalér brøkverdien til meter ved å multiplisere med bufferavstanden
  var scaledDistanceWithInfrastructure = ee.Number(distanceWithInfrastructure).multiply(bufferDistance);
  var scaledDistanceWithoutInfrastructure = ee.Number(distanceWithoutInfrastructure).multiply(bufferDistance);

  // Beregn kvotienten (forholdet mellom avstand uten infrastruktur og avstand med infrastruktur)
  var kvotient = scaledDistanceWithoutInfrastructure.divide(scaledDistanceWithInfrastructure);

  var properties = {
    'distinfra_meters': scaledDistanceWithInfrastructure,
    'distnoinfra_meters': scaledDistanceWithoutInfrastructure,
    'kvotient': kvotient
  };

  return feature.set(properties);
};

// Velg områder med myr (myrinnhold > 80%) og bruk dette som geometri for analysen
var omraderMedMyr = myr.gt(0.8).selfMask();

// Hent geometriene for hvert område med myr
var omraderGeometries = omraderMedMyr.reduceToVectors({
  geometry: geometry,
  scale: 10,  // Oppløsningen til datasettet (meters per piksel)
  maxPixels: 1e13
});

// Bruk funksjonen calculateMeanDistance for å beregne gjennomsnittlig avstand til infrastruktur for hvert område med myr
var omraderMedGjennomsnittligAvstand = omraderGeometries.map(calculateMeanDistance);

// Legg til lag som viser myrpolygonene og lag med gjennomsnittlig avstand til infrastruktur på kartet
Map.addLayer(omraderMedMyr, { min: 0, max: 1, palette: ['white', 'green'] }, 'Myrpolygoner');

// Vis scenariet "Konnektivitet med infrastruktur" på kartet
var visParams = {
  min: 0,
  max: 1, // Juster dette basert på din infrastrukturindeksens min og maks
  palette: ['blue', 'yellow', 'red'],
};
Map.addLayer(omraderMedGjennomsnittligAvstand, visParams, 'Gjennomsnittlig avstand til infrastruktur');

// Eksporter resultatene som en Shapefile til Google Drive
Export.table.toDrive({
  collection: omraderMedGjennomsnittligAvstand,
  description: 'Konnektivitet_med_infrastruktur',
  fileFormat: 'SHP',
});

Reply:
This was indeed a much richer code block. Thank you @VegarBakkestuen 

I can for example see that you calculate distances from the polygon centroids, and that you use level 3 for the infrastructure index as a threshold for when it should act as a barrier. Did I understand that correctly? These are thing that are nice to write up in plain text in the chapter as well. 

Since I'm not that familiar with java (or is it javaScript?) I cannot make out what `distinfra_meters` and `distnoinfra_meters` actually is. Can you please explain this in text as well? I assumed they where both distances to other mire polygins, but then I'm not sure how the infrastructure comes into this.

Finally, what does it mean to _redusere regionen rundt hvert myrpolygon_.

I will add this code into the pull request now.

Reply:
Btw, to edit the chapter yourself directly in the web browser (assuming you dont use Rstudio or have github desktop installed), go to the [github repo and the code tab](https://github.com/NINAnor/ecosystemCondition), then swith to the branch we are working on (the current branch for the connectivity review is called _Alk connectivity review_) and open the DRAFT_connectivity.Rmd file:
![image](https://github.com/NINAnor/ecosystemCondition/assets/33656949/d25772aa-13f1-4649-9ddd-252e2b6b3da1)

Click _edit_: 
![image](https://github.com/NINAnor/ecosystemCondition/assets/33656949/a4618bd5-1397-454c-8cb8-0c59dafc3eac)


and make you changes. Then find the big green _commit_ button and commit directly to the branch. This will update the branch and the pull request is updated automatically. 

Happy to help if you get stuck :)